apps/student-app/src/app/student-app.tsx

```diff
@@ -0,0 +1,379 @@
+          </span>
+        ) : null}
+        <div id="word-list-select">
+          <form id="course-select-form" onSubmit={handleSubmitCourse}>
+            {error === 'error-word-list' ? (
+              <span id="errorCourse" className="error">
+                Failed to load course guid.
+              </span>
+            ) : null}
+            <label>Select Word List: </label>
+            <ul id="word-list-list">
+              {populate_word_list_items()}
+              </ul>
+            <button type="submit" id="submit-word-list-btn">
+              Submit
+            </button>
+          </form>
+          <button id="back-to-course-btn" onClick={handleBackToCourse}>
+            Back
+          </button>
+        </div>
+      </div>
+
+      <div
+        id="game-screen"
+        className={screen === StudentState.game_state ? 'content' : 'hidden-content'}
+      >
+        <form id="course-select-form" >
+          <h1 id="word-hint-header">{current_word?.hint}</h1>
+          <label>Enter Word: </label>
+          <input
+            type="text"
+            id="word-test"
+            name="word-test"
+            placeholder="word"
+          />
+          <br></br>
+          <button disabled={false}>Check</button>
+          <div id="word-answer">
+            <h2
+              id="word-answer-header"
+            >
+              {current_word?.word}
+            </h2>
+          </div>
+          <div id="score">Score: {score}</div>
+          <div id="time-left">
+            {time_remaining} {time_remaining === 1 ? 'second' : 'seconds'}
+          </div>
+        </form>
+      </div>
+
+      <div
+        id="game-screen"
+        className={screen === StudentState.score_state ? 'content' : 'hidden-content'}
+      >
+        <h1 id="score-header">Congrats!</h1>
+        <div>
+          You got {score} of {words_in_game.length}!
+        </div>
+        <button id="back-to-course-btn" onClick={handleResetGame}>
+          Replay
+        </button>
+        <button id="game-words-btn" onClick={handleToGameWords}>
+          Review Words
+        </button>
+        <br></br>
+
+        <button id="back-to-word-list-btn" onClick={handleBackToWordLists}>
+          New Wordlist
+        </button>
+        <button id="back-to-course-btn" onClick={handleBackToCourse}>
+          Change Course
+        </button>
+      </div>
+
+      <div
+        id="game-screen"
+        className={screen === StudentState.review_words_state ? 'content' : 'hidden-content'}
+      >
+        <h1>Vocab Game</h1>
+        <ul>
+          {get_words_from_game()}
+        </ul>
+        <button onClick={handleBackToScore}>Back to Score</button>
+      </div>
+    </>
+  );
+}
+
+// Stub of GET request for data
+function submitCourse(code: string): Promise<Data> {
+  // Pretend it's hitting the network.
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      let shouldError = code.toLowerCase() !== 'qwerty';
```

Comment:
`let` to `const` cause it is never re-assigned. ---- **but** don't bother changing, this is temp code

In fact your default should always be `const`, and if you find yourself using a `let`, it's a good sign to pause and think about how you are writing the thing to see if that is really necessary.

Reply:
in this context, let probably means put it into the react state if it is going to change.

Reply:
Lets just get rid of the boolean and move the condition into the conditional.